src/ide/semantic_highlighting/token_kind.rs

```diff
@@ -271,3 +274,18 @@ fn get_resultants_and_closest_terminals(
         })
         .collect()
 }
+
+/// Checks whether the given node is an inline macro invocation and not just the simple path segment.
+fn is_inline_macro(db: &AnalysisDatabase, node: SyntaxNode) -> bool {
+    if node.kind(db) == SyntaxKind::PathSegmentSimple
+        && let Some(parent) = node.parent(db)
+        && parent.kind(db) == SyntaxKind::ExprPathInner
+        && let Some(grandparent) = parent.parent(db)
+        && grandparent.kind(db) == SyntaxKind::ExprPath
+    {
+        if let Some(great_grandparent) = grandparent.parent(db) {
+            return great_grandparent.kind(db) == SyntaxKind::ExprInlineMacro;
+        }
+    }
+    false
```

Comment:
```suggestion
    if let Some(path_node) = node.ancestor_of_kind(SyntaxKind::ExprPath) && let Some(maybe_macro) = path_node.parent()
    {
        let kind = maybe_macro.kind(db);
        return kind == SyntaxKind::ExprInlineMacro || kind == SyntaxKind::ItemInlineMacro;
    }
    false
```

@piotmag769 version is incorrect as it will catch whole 
```
macro_call!(args)
```
instead of `macro_call` only.

Yours is to precise, if syntax definition changes even slightly here it will break, try to avoid this precision if possible to do without ambiguity